src/UCI.cpp

```diff
@@ -309,65 +335,82 @@ void UCI::loop(const std::string& start) {
       else if (token == "train")   generate_training_games(is);
       else if (token == "bench")   bench();
       else if (token == "d" || token == "showboard") {
-		  std::stringstream ss;
-		  ss << bh.cur();
-		  myprintf_so("%s\n", ss.str().c_str());
-	  }
-	  else if (token == "showfen") {
-	      std::stringstream ss;
-		  ss << bh.cur().fen();
-		  myprintf_so("%s\n", ss.str().c_str());
-	  }
-	  else if (token == "showgame") {
-		  std::string result;
-		  for (const auto &p : bh.positions) {
-			  if (result == "") {
-			      result = " "; // first position has no move
-			  } else {
-				  result += UCI::move(p.get_move()) + " ";
-			  }
-		  }
-		  myprintf_so("position startpos%s\n", result.c_str());
-	  }
-	  else if (token == "showpgn") myprintf_so("%s\n", bh.pgn().c_str());
-	  else if (token == "undo") myprintf_so(bh.undo_move() ? "Undone\n" : "At first move\n");
-	  else if (token == "usermove" || token == "play") {
-		  std::string ms; is >> ms;
-		  Move m = UCI::to_move(bh.cur(), ms);
-		  if (m == MOVE_NONE) m = bh.cur().san_to_move(ms);
-		  if (m != MOVE_NONE) {
-			  bh.do_move(m);
-			  myprintf_so("usermove %s\n", UCI::move(m).c_str());
-		  }
-		  else {
-			  myprintf_so("Illegal move: %s\n", ms.c_str());
-		  }
-	  }
-	  else if (UCI::to_move(bh.cur(), token) != MOVE_NONE) {
-		  Move m = UCI::to_move(bh.cur(), token);
-		  bh.do_move(m);
-		  myprintf_so("usermove %s\n", UCI::move(m).c_str());
-	  }
-		  //else if (token == "eval")  sync_cout << Eval::trace(pos) << sync_endl;
-		  else if (token != "quit") {
-		  myprintf_so("Unknown command: %s\n", cmd.c_str());
-	  }
+          std::stringstream ss;
+          ss << bh.cur();
+          myprintf("%s\n", ss.str().c_str());
+      }
+      else if (token == "testgo") {
```

Comment:
Maybe users can do "ucinewgame" and "go"? Then we don't need a separate command. Or is there another reason for this command?